File: 2232-adding-spaces-to-a-string/2232-adding-spaces-to-a-string.cs
public class Solution {
    public string AddSpaces(string s, int[] spaces) {
        char[] newString = new char[s.Length + spaces.Length];
        int addedSpace = 0;
        int i = -1;
        int spaceIndex = 0;

        while (spaceIndex<spaces.Length && ++i<s.Length) {
            if (i == spaces[spaceIndex]) { 
                newString[i + addedSpace] = ' ';
                addedSpace++;
                spaceIndex++;
            }
            newString[i+addedSpace] = s[i];        
        }
        while(++i<s.Length)
            newString[i+addedSpace] = s[i];
        
        return new string(newString);
    }
}

File: 2232-adding-spaces-to-a-string/2232-adding-spaces-to-a-string.go
func addSpaces(s string, spaces []int) string {
    a := make([]uint8, 0, len(s) + len(spaces))
    for i, j := 0, 0; i < len(s); i++ {
        if j < len(spaces) && spaces[j] == i {
            a = append(a, ' ')
            j++
        }
        a = append(a, s[i])
    }
    return string(a)
}